File: Stage18/Java/Problem1037.java
package Stage18.Java;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class Problem1037 {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        
        int N = Integer.parseInt(br.readLine());
        StringTokenizer st = new StringTokenizer(br.readLine());

        long min = Long.MAX_VALUE;
        long max = Long.MIN_VALUE;

        for (int i = 0; i < N; i++) {
            long d = Long.parseLong(st.nextToken());
            if (d < min) min = d;
            if (d > max) max = d;
        }

        long result;
        if (N == 1) result = min * min;
        else result = min * max;

        System.out.println(result);
        br.close();
    }
}

File: Stage18/C/Problem1037.c
#include <stdio.h>

int main() {
    int N, x;
    int min = 2147483647;
    int max = -2147483648;

    scanf("%d", &N);

    while (N--) {
        scanf("%d", &x);
        if (x < min) min = x;
        if (x > max) max = x;
    }

    printf("%lld\n", 1LL * min * max);
    return 0;
}